test/types/populate.test.ts

```diff
@@ -339,3 +339,34 @@ function gh12136() {
   }
 
 }
+
+async function gh13070() {
+  interface IParent {
+    name: string;
+    child: Types.ObjectId;
+  }
+  interface IChild {
+    name: string;
+    parent: Types.ObjectId;
+  }
+
+  const parentSchema = new Schema(
+    {
+      name: { type: String, required: true },
+      child: { type: Schema.Types.ObjectId, ref: 'Child', required: true }
+    });
+
+  const childSchema = new Schema(
+    {
+      name: { type: String, required: true },
+      parent: { type: Schema.Types.ObjectId, ref: 'Parent', required: true }
+    });
+
+  const parent = model<IParent>('Parent', parentSchema);
+  const child = model<IChild>('Child', childSchema);
+
+  const doc = await parent.findOne();
+  await child.populate<{child: IChild}>(doc, 'child');
```

Comment:
Didn't actually test that the result doc is correctly populated. Missed this: https://github.com/Automattic/mongoose/blob/7d2c511475fb4a36d8ffcdd30739af45cdde4d4e/test/types/populate.test.ts#L126